### PR TITLE
fix stringio issue in py3.5.2

### PR DESCRIPTION
**Pull request by guyromm**
this is a fix due to upload_template() breaking. it seems that put no longer works with stringio?

### CODE REVIEW

File: fabric/contrib/files.py [hunk 1]
```diff
@@ -168,9 +169,12 @@ def upload_template(filename, destination, context=None, use_jinja=False,
     if six.PY3 is True and isinstance(text, bytes):
         text = text.decode('utf-8')
 
+    tf = tempfile.NamedTemporaryFile(mode='w',delete=False)
+    tf.write(text) ; tf.close()
```
Comment (cclauss) [inline fabric/contrib/files.py]:
```python
with tempfile.NamedTemporaryFile(mode='w', delete=False) as tf:
    tf.write(text)
```

**Comment by guyromm** [comment]
please re-review

**Comment by mathiasertl** [comment]
This looks to me like it might be a bytes/str issue. Can you provide a piece of code that triggers the error?

**Comment by mathiasertl** [comment]
@guyromm Can you please rebase on the current master, which includes some travis updates and test fixes so they pass with py3.

I'm still hoping for a showcase that triggers the error ;-)

**Comment by mathiasertl** [comment]
Since there still is no testcase for this, closing this PR.

**Pull request closed without merging**